src/filesys.cpp

```diff
@@ -740,5 +740,26 @@ bool Rename(const std::string &from, const std::string &to)
 	return rename(from.c_str(), to.c_str()) == 0;
 }
 
+bool bad_char(const char c){
+	if (c >= 0 && c <= 31)
+		return true;
+	if (bad_chars.find_first_of(c) != std::string::npos)
+		return true;
+	if (c > 255)
+		return true;
+	return false;
+}
+
+void FixBadFilename(std::string &input){
```

Comment:
I'd prefer this function to have no side affects - it's conceptually simpler. This code won't be called much so there's no need for optimisation

```cpp
std::string SanitizeFilename(std::string input)
{
	input.erase(remove_if(input.begin(), input.end(), bad_char), input.end());
	input.erase(input.find_last_not_of(" ") + 1);
	for (unsigned int a = 0; a < sizeof(bad_names) / sizeof(bad_names[0]); a++) {
		if (input == bad_names[a]) {
			input = input + "_";
			break;
		}
	}
	return input;
}
```